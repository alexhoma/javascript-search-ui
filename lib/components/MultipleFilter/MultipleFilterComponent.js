"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
exports.__esModule = true;
var preact_1 = require("preact");
var MultipleFilterActions_1 = require("./MultipleFilterActions");
var Helpers_1 = require("./Helpers");
var Template_1 = require("../Template");
var ShowMoreComponent_1 = require("./ShowMoreComponent");
var defaultTemplates_1 = require("./defaultTemplates");
var apisearch_1 = require("apisearch");
/**
 * Filter Component
 */
var MultipleFilterComponent = /** @class */ (function (_super) {
    __extends(MultipleFilterComponent, _super);
    /**
     * Constructor
     */
    function MultipleFilterComponent() {
        var _this = _super.call(this) || this;
        /**
         * Handle click
         *
         * @param selectedFilter
         */
        _this.handleClick = function (selectedFilter) {
            var props = _this.props;
            var environmentId = props.environmentId;
            var filterName = props.filterName;
            var filterField = props.filterField;
            var aggregationField = props.aggregationField;
            var applicationType = props.applicationType;
            var sortBy = props.sortBy;
            var ranges = props.ranges;
            var labels = props.labels;
            var fetchLimit = props.fetchLimit;
            var repository = props.repository;
            var currentQuery = props.store.getCurrentQuery();
            var aggregation = props.store.getCurrentResult().getAggregation(filterName);
            var selectedFilterAsString = String(selectedFilter);
            var currentActiveFilterValues = aggregation instanceof apisearch_1.ResultAggregation
                ? Object.values(aggregation.getActiveElements())
                : [];
            var valuesAsString = currentActiveFilterValues.map(function (element) {
                return String(element);
            });
            /**
             * Dispatch filter action
             */
            MultipleFilterActions_1.filterAction(environmentId, currentQuery, repository, filterName, filterField, (aggregationField
                ? aggregationField
                : filterField), Helpers_1.manageCurrentFilterItems(selectedFilterAsString, valuesAsString), applicationType, sortBy, fetchLimit, ranges, labels);
        };
        /**
         * Handle show more
         */
        _this.handleShowMore = function () {
            var viewLimit = _this.state.aggregations.length;
            _this.setState(function (prevState) {
                return { viewLimit: viewLimit };
            });
        };
        /**
         * Handle show less
         */
        _this.handleShowLess = function () {
            var viewLimit = _this.props.viewLimit;
            _this.setState(function (prevState) {
                return { viewLimit: viewLimit };
            });
        };
        _this.state = {
            viewLimit: 0,
            aggregations: []
        };
        return _this;
    }
    /**
     * Components will mount
     */
    MultipleFilterComponent.prototype.componentWillMount = function () {
        var props = this.props;
        var environmentId = props.environmentId;
        var filterName = props.filterName;
        var filterField = props.filterField;
        var aggregationField = props.aggregationField;
        var applicationType = props.applicationType;
        var sortBy = props.sortBy;
        var ranges = props.ranges;
        var fetchLimit = props.fetchLimit;
        var viewLimit = props.viewLimit;
        var currentQuery = props.store.getCurrentQuery();
        /**
         * Set view items limit
         */
        var isViewLimitProperlySet = (viewLimit && viewLimit < fetchLimit);
        this.setState(function (prevState) {
            return {
                viewLimit: (isViewLimitProperlySet)
                    ? viewLimit
                    : fetchLimit
            };
        });
        /**
         * Dispatch action
         */
        MultipleFilterActions_1.aggregationSetup(environmentId, currentQuery, filterName, (aggregationField
            ? aggregationField
            : filterField), applicationType, sortBy, fetchLimit, ranges);
    };
    /**
     * Component will receive props
     *
     * @param props
     */
    MultipleFilterComponent.prototype.componentWillReceiveProps = function (props) {
        var filterName = props.filterName;
        if (props.store.getCurrentResult() == null) {
            this.setState(function (prevState) {
                return {
                    aggregations: []
                };
            });
            return;
        }
        var aggregation = props.store.getCurrentResult().getAggregation(filterName);
        if (typeof aggregation.getCounters === "function") {
            /**
             * Getting aggregation from aggregations
             */
            var counters = aggregation.getCounters();
            var countersAsArray = Object.values(counters);
            var aggregations_1 = props.activeFirst
                ? __spreadArrays(countersAsArray.filter(function (counter) {
                    return true === counter.isUsed();
                }), countersAsArray.filter(function (counter) {
                    return (false === counter.isUsed() ||
                        null === counter.isUsed());
                })) : countersAsArray;
            this.setState(function (prevState) {
                return {
                    aggregations: aggregations_1
                };
            });
        }
    };
    /**
     * Render
     *
     * @return {any}
     */
    MultipleFilterComponent.prototype.render = function () {
        var _this = this;
        var props = this.props;
        var viewLimit = props.viewLimit;
        var fetchLimit = props.fetchLimit;
        var containerClassName = props.classNames.container;
        var topClassName = props.classNames.top;
        var itemsListClassName = props.classNames.itemsList;
        var itemClassName = props.classNames.item;
        var activeClassName = props.classNames.active;
        var showMoreContainerClassName = props.classNames.showMoreContainer;
        var topTemplate = props.template.top;
        var itemTemplate = props.template.item;
        var showMoreTemplate = props.template.showMore;
        var showLessTemplate = props.template.showLess;
        var formatData = props.formatData;
        var labels = Object.keys(props.ranges).length > 0
            ? props.ranges
            : props.labels;
        /**
         * Get aggregation items
         */
        var allItems = this.state.aggregations;
        var allItemsLength = allItems.length;
        var items = allItems.slice(0, this.state.viewLimit);
        var that = this;
        if (allItems.length == 0) {
            return null;
        }
        /**
         * Check available view limit
         */
        var isViewLimitProperlySet = (viewLimit &&
            viewLimit < fetchLimit &&
            allItemsLength > viewLimit);
        return (preact_1.h("div", { className: "as-multipleFilter " + containerClassName },
            preact_1.h(Template_1["default"], { template: topTemplate, className: "as-multipleFilter__top " + topClassName, dictionary: this.props.dictionary }),
            preact_1.h("div", { className: "as-multipleFilter__itemsList " + itemsListClassName }, items.map(function (item) {
                var values = item.getValues();
                values.name = labels[values.name] ? labels[values.name] : values.name;
                var uid = Math.floor(Math.random() * 10000000000);
                var reducedTemplateData = {
                    n: item.getN(),
                    isActive: item.isUsed(),
                    values: values,
                    uid: uid
                };
                var formattedTemplateData = formatData(reducedTemplateData);
                return (preact_1.h("div", { className: "as-multipleFilter__item " +
                        (itemClassName + " ") +
                        ("" + ((item.used) ? activeClassName : '')), onClick: function (e) {
                        e.stopPropagation();
                        e.preventDefault();
                        that.handleClick(item.values.id);
                    } },
                    preact_1.h(Template_1["default"], { template: itemTemplate, data: formattedTemplateData, dictionary: _this.props.dictionary })));
            })),
            (isViewLimitProperlySet)
                ? preact_1.h(ShowMoreComponent_1["default"], { allItemsLength: allItemsLength, currentLimit: this.state.viewLimit, handleShowMore: this.handleShowMore, handleShowLess: this.handleShowLess, showMoreContainerClassName: showMoreContainerClassName, showMoreTemplate: showMoreTemplate, showLessTemplate: showLessTemplate, dictionary: this.props.dictionary }) : null));
    };
    return MultipleFilterComponent;
}(preact_1.Component));
MultipleFilterComponent.defaultProps = {
    aggregationField: null,
    applicationType: 8,
    fetchLimit: 10,
    viewLimit: null,
    sortBy: ['_term', 'desc'],
    ranges: {},
    labels: {},
    classNames: {
        container: '',
        top: '',
        itemsList: '',
        item: '',
        active: 'as-multipleFilter__item--active',
        showMoreContainer: ''
    },
    template: {
        top: null,
        item: defaultTemplates_1.defaultItemTemplate,
        showMore: '+ Show more',
        showLess: '- Show less'
    },
    formatData: function (data) { return data; },
    activeFirst: true
};
exports["default"] = MultipleFilterComponent;
