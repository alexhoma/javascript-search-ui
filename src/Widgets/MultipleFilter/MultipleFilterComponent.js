/**
 * @jsx h
 */
import { h, Component } from 'preact';
import {aggregationSetup, filterAction} from "./multipleFilterActions";
import Template from "../Template";
import {
    aggregationsObjectToArray, manageCurrentFilterItems,
    simpleObjectToArray
} from "./helpers";

/**
 * Filter Component
 */
class MultipleFilterComponent extends Component {
    constructor() {
        super();
        this.state = {
            activeAggregations: [],
            currentAggregations: []
        }
    }

    componentWillMount() {
        const {
            field: filterField,
            name: filterName,
            currentQuery
        } = this.props;

        /**
         * Dispatach action
         */
        aggregationSetup(
            {
                filterField,
                filterName
            },
            currentQuery
        )
    }

    componentWillReceiveProps(props) {
        const {
            name: filterName,
            data: {
                aggregations: {
                    aggregations
                }
            }
        } = props;

        if (typeof aggregations[filterName] !== 'undefined') {
            let aggregation = aggregations[filterName];
            let counters = (aggregation.counters) ? aggregation.counters : {};
            let aggregationsArray = aggregationsObjectToArray(counters);

            this.setState({
                /**
                 * Current used aggregations
                 */
                activeAggregations: aggregationsArray.filter(
                    item => item.used
                ),
                /**
                 * Current inactive aggregations
                 */
                currentAggregations: aggregationsArray.filter(
                    item => null === item.used
                )
            })
        }
    }

    handleClick = (selectedFilter) => {
        const {
            name: filterName,
            field: filterField,
            currentQuery,
            client,
            data: {
                aggregations: {
                    aggregations
                }
            }
        } = this.props;

        let activeElements = aggregations[filterName].active_elements;
        let currentActiveFilterValues = (typeof activeElements !== 'undefined')
            ? simpleObjectToArray(activeElements)
            : []
        ;

        /**
         * Dispatch action
         */
        filterAction(
            {
                filterName,
                filterField,
                filterValues: manageCurrentFilterItems(
                    selectedFilter,
                    currentActiveFilterValues
                )
            },
            currentQuery,
            client
        );
    };

    render() {
        const {
            classNames: {
                container: containerClassName,
                top: topClassName,
                itemsList: itemsListClassName,
                item: itemClassName
            },
            template: {
                top: topTemplate,
                item: itemTemplate
            }
        } = this.props;

        /**
         * Get aggregation items
         */
        const items = [
            ...this.state.activeAggregations,
            ...this.state.currentAggregations
        ];

        return (
            <div className={`asui-multipleFilter ${containerClassName}`}>
                <Template
                    template={topTemplate}
                    className={`asui-multipleFilter--top ${topClassName}`}
                />

                <div className={itemsListClassName}>
                {items.map(item => {
                    const reducedTemplateData = {
                        n: parseInt(item.n).toLocaleString('de-DE'),
                        isActive: item.used,
                        values: item.values
                    };

                    return (
                        <div
                            className={`asui-multipleFilter--item ${itemClassName}`}
                            onClick={() => this.handleClick(item.__key)}
                        >
                            <Template
                                template={itemTemplate}
                                data={reducedTemplateData}
                            />
                        </div>
                    )
                })}
                </div>
            </div>
        )
    }
}

MultipleFilterComponent.defaultProps = {
    type: 'FILTER_MUST_ALL',
    limit: 10,
    classNames: {
        container: '',
        top: '',
        itemList: '',
        item: ''
    },
    template: {
        top: null,
        item: null
    }
};

export default MultipleFilterComponent;