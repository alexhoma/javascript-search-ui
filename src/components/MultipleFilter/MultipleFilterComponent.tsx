import { h, Component } from 'preact';

import * as cloneDeep from 'clone-deep';
import { aggregationSetup, filterAction } from "./MultipleFilterActions";
import { manageCurrentFilterItems } from "./Helpers";

import Template from "../Template";
import ShowMoreComponent from "./ShowMoreComponent";
import {defaultItemTemplate} from "./defaultTemplates";
import {MultipleFilterProps} from "./MultipleFilterProps";
import {MultipleFilterState} from "./MultipleFilterState";
import {Counter} from 'apisearch';
import {ResultAggregation} from 'apisearch';

/**
 * Filter Component
 */
class MultipleFilterComponent extends Component<MultipleFilterProps, MultipleFilterState> {

    /**
     * Constructor
     */
    constructor() {
        super();
        this.state = {
            viewLimit: 0,
            aggregations: []
        }
    }

    /**
     * Components will mount
     */
    componentWillMount() {

        const props = this.props;
        const environmentId = props.environmentId;
        const filterName = props.filterName;
        const filterField = props.filterField;
        const aggregationField = props.aggregationField;
        const applicationType = props.applicationType;
        const sortBy = props.sortBy;
        const ranges = props.ranges;
        const fetchLimit = props.fetchLimit;
        const viewLimit = props.viewLimit;
        const currentQuery = props.currentQuery;

        /**
         * Set view items limit
         */
        const isViewLimitProperlySet = (viewLimit && viewLimit < fetchLimit);
        this.setState(prevState => {
            return {
               viewLimit: (isViewLimitProperlySet)
                   ? viewLimit
                   : fetchLimit
           };
        });

        /**
         * Dispatch action
         */
        aggregationSetup(
            environmentId,
            currentQuery,
            filterName,
            (
                aggregationField
                    ? aggregationField
                    : filterField
            ),
            applicationType,
            sortBy,
            fetchLimit,
            ranges
        );
    }

    /**
     * Component will receive props
     *
     * @param props
     */
    componentWillReceiveProps(props) {

        const filterName = props.filterName;
        if (props.currentResult == null) {
            this.setState(prevState => {
                return {
                    aggregations: []
                };
            });

            return;
        }

        const aggregation = props.currentResult.getAggregation(filterName);

        if (typeof aggregation.getCounters === "function") {

            /**
             * Getting aggregation from aggregations
             */
            let counters = aggregation.getCounters();
            let countersAsArray:Counter[] = Object.values(counters);

            const aggregations = props.activeFirst
                ? [
                    ...countersAsArray.filter(
                        counter =>
                            true === counter.isUsed()
                    ),
                    ...countersAsArray.filter(
                        counter =>
                            (
                                false === counter.isUsed() ||
                                null === counter.isUsed()
                            )
                    )
                ]
                : countersAsArray;

            this.setState(prevState => {
                return {
                    aggregations: aggregations
                };
            });
        }
    }

    /**
     * Handle click
     *
     * @param selectedFilter
     */
    handleClick = (selectedFilter) => {
        const props = this.props;
        const environmentId = props.environmentId;
        const filterName = props.filterName;
        const filterField = props.filterField;
        const aggregationField = props.aggregationField;
        const applicationType = props.applicationType;
        const sortBy = props.sortBy;
        const ranges = props.ranges;
        const labels = props.labels;
        const fetchLimit = props.fetchLimit;
        const repository = props.repository;
        const currentQuery = props.currentQuery;
        const aggregation = props.currentResult.getAggregation(filterName);
        const selectedFilterAsString = String(selectedFilter);
        const currentActiveFilterValues = aggregation instanceof ResultAggregation
            ? Object.values(aggregation.getActiveElements())
            : [];

        const valuesAsString = currentActiveFilterValues.map(element => {
            return String(element);
        });

        /**
         * Dispatch filter action
         */
        filterAction(
            environmentId,
            currentQuery,
            repository,
            filterName,
            filterField,
            (
                aggregationField
                    ? aggregationField
                    : filterField
            ),
            manageCurrentFilterItems(
                selectedFilterAsString,
                valuesAsString
            ),
            applicationType,
            sortBy,
            fetchLimit,
            ranges,
            labels
        );
    };

    /**
     * Handle show more
     */
    handleShowMore = () => {

        const viewLimit = this.state.aggregations.length;

        this.setState(prevState => {
            return {viewLimit};
        });
    };

    /**
     * Handle show less
     */
    handleShowLess = () => {
        const viewLimit = this.props.viewLimit;
        this.setState(prevState => {
            return {viewLimit};
        });
    };

    /**
     * Render
     *
     * @return {any}
     */
    render() {
        const props = this.props;
        const viewLimit = props.viewLimit;
        const fetchLimit = props.fetchLimit;

        const containerClassName = props.classNames.container;
        const topClassName = props.classNames.top;
        const itemsListClassName = props.classNames.itemsList;
        const itemClassName = props.classNames.item;
        const activeClassName = props.classNames.active;
        const showMoreContainerClassName = props.classNames.showMoreContainer;

        const topTemplate = props.template.top;
        const itemTemplate = props.template.item;
        const showMoreTemplate = props.template.showMore;
        const showLessTemplate = props.template.showLess;

        const formatData = props.formatData;
        const labels = Object.keys(props.ranges).length > 0
            ? props.ranges
            : props.labels;

        /**
         * Get aggregation items
         */
        const allItems = this.state.aggregations;
        const allItemsLength = allItems.length;
        const items = allItems.slice(0, this.state.viewLimit);

        if (allItems.length == 0) {
            return null;
        }

        /**
         * Check available view limit
         */
        const isViewLimitProperlySet = (
            viewLimit &&
            viewLimit < fetchLimit &&
            allItemsLength > viewLimit
        );

        return (
            <div className={`as-multipleFilter ${containerClassName}`}>
                <Template
                    template={topTemplate}
                    className={`as-multipleFilter__top ${topClassName}`}
                />

                <div className={`as-multipleFilter__itemsList ${itemsListClassName}`}>
                {items.map(item => {
                    const values = item.getValues();
                    values.name = labels[values.name] ? labels[values.name] : values.name;
                    const uid = Math.floor(Math.random() * 10000000000);
                    const reducedTemplateData = {
                        n: item.getN(),
                        isActive: item.isUsed(),
                        values: values,
                        uid: uid
                    };
                    const formattedTemplateData = formatData(reducedTemplateData);
                    return (
                        <div
                            className={
                                `as-multipleFilter__item ` +
                                `${itemClassName} ` +
                                `${(item.used) ? activeClassName : ''}`
                            }
                            onClick={() => this.handleClick(item.values.id)}
                        >
                            <Template
                                template={itemTemplate}
                                data={formattedTemplateData}
                            />
                        </div>
                    )
                })}
                </div>

                {(isViewLimitProperlySet)
                    ? <ShowMoreComponent
                        allItemsLength={allItemsLength}
                        currentLimit={this.state.viewLimit}
                        handleShowMore={this.handleShowMore}
                        handleShowLess={this.handleShowLess}
                        showMoreContainerClassName={showMoreContainerClassName}
                        showMoreTemplate={showMoreTemplate}
                        showLessTemplate={showLessTemplate}
                    /> : null
                }
            </div>
        )
    }
}

MultipleFilterComponent.defaultProps = {
    aggregationField: null,
    applicationType: 8, // FILTER_MUST_ALL
    fetchLimit: 10,
    viewLimit: null,
    sortBy: ['_term', 'desc'],
    ranges: {},
    labels: {},
    classNames: {
        container: '',
        top: '',
        itemsList: '',
        item: '',
        active: 'as-multipleFilter__item--active',
        showMoreContainer: ''
    },
    template: {
        top: null,
        item: defaultItemTemplate,
        showMore: '+ Show more',
        showLess: '- Show less'
    },
    formatData: data => data,
    activeFirst: true
};

export default MultipleFilterComponent;
