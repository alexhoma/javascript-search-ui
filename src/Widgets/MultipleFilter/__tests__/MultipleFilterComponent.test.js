/** @jsx h */
import {h} from 'preact';
import {deep} from 'preact-render-spy';
import {initialDataProvider, updatedDataProvider} from './dataProvider.js';
import MultipleFilterComponent from './../MultipleFilterComponent';

describe('<MultipleFilterComponent />', () => {
    const itemTemplate = `
        <a class="aggregation {{#isActive}}is-active{{/isActive}}">
            <input type="checkbox" id="filter_{{values.id}}" {{#isActive}}checked="checked"{{/isActive}}>
            <label for="filter_{{values.id}}">{{values.name}} ({{n}})</label>
        </a>           
    `;
    const defaultProps = {
        target: '#some-target',
        name: 'year',
        filterField: 'year_id',
        template: {
            item: itemTemplate
        }
    };
    const customProps = {
        target: '#some-target',
        name: 'year',
        filterField: 'year_id',
        aggregationField: 'year_data',
        applicationType: 4,
        fetchLimit: 6,
        viewLimit: 3,
        showMoreActive: true,
        classNames: {
            container: 'container-custom-class',
            top: 'top-custom-class',
            itemsList: 'itemsList-custom-class',
            item: 'item-custom-class',
            showMoreContainer: 'showMoreContainer-custom-class'
        },
        template: {
            top: 'Filter top header template:',
            item: itemTemplate,
            showMore: '<b>+ Show more</b>',
            showLess: '<b>- Show less</b>'
        }
    };

    it('should render with default props', () => {
        /**
         * Rendering component with default props (dirty)
         * Updating Store query on componentWillMount()
         * Re-rendering component with aggregations
         */
        let tree = getDirtyMultipleFilterComponent(defaultProps);
        expect(tree.component().props).toMatchSnapshot();

        tree = updateMultipleFilterComponent(tree, defaultProps, initialDataProvider);
        expect(tree).toMatchSnapshot();
    });
    it('should render with custom props, and re-render with aggregations', () => {
        /**
         * Rendering component with all custom props (dirty)
         * Updating Store query on componentWillMount()
         * Re-rendering component with aggregations
         */
        let tree = getDirtyMultipleFilterComponent(customProps);
        expect(tree.component().props).toMatchSnapshot();

        tree = updateMultipleFilterComponent(tree, customProps, initialDataProvider);
        expect(tree).toMatchSnapshot();
    });
    it('should handleCLick() given a selected aggregation and set it as active', () => {
        /**
         * Create and mount a component
         */
        const mock = jest.fn();
        let tree = getDirtyMultipleFilterComponent(customProps);
        tree = updateMultipleFilterComponent(tree, customProps, initialDataProvider);

        tree.find('.asui-multipleFilter--item')[1].attributes.onClick = mock;
        tree.find('.asui-multipleFilter--item').at(1).simulate('click');

        tree = updateMultipleFilterComponent(tree, customProps, updatedDataProvider);

        expect(mock).toBeCalled();
        expect(tree).toMatchSnapshot();
    });
    it('should handleShowMore() and handleShowLess()', () => {
        /**
         * Create and mount a component
         */
        let tree = getDirtyMultipleFilterComponent(customProps);
        tree = updateMultipleFilterComponent(tree, customProps, initialDataProvider);

        /**
         * Click on show more function
         */
        tree.find('.asui-showMore').at(0).simulate('click');
        tree = updateMultipleFilterComponent(tree, customProps, initialDataProvider);

        /**
         * It should show the third and last aggregation item
         * plus a ShowLess button
         */
        expect(tree).toMatchSnapshot();

        /**
         * Click on show less function
         */
        tree.find('.asui-showMore').at(0).simulate('click');
        tree = updateMultipleFilterComponent(tree, customProps, initialDataProvider);

        /**
         * It should hide the third and last aggregation item
         * plus a ShowMore button
         */
        expect(tree).toMatchSnapshot();
    });

    const getDirtyMultipleFilterComponent = (props) => {
        MultipleFilterComponent.prototype.componentWillMount = jest.fn();
        const tree = deep(
            <MultipleFilterComponent
                dirty={true}
                {...props}
            />
        );
        /**
         * Mocking componentWillMount functionality
         */
        tree.setState({limit: props.limit});

        return tree;
    };
    const updateMultipleFilterComponent = (tree, props, data) => {
        tree.render(
            <MultipleFilterComponent
                dirty={false}
                {...props}
                data={data}
            />
        );

        return tree;
    }
});