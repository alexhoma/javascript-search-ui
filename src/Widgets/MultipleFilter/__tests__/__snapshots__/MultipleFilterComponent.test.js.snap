// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleFilterComponent /> should handleCLick() given a selected aggregation and set it as active 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-multipleFilter container-custom-class">
  <div class="asui-multipleFilter--top top-custom-class">Filter top header template:</div>
  <div class="asui-multipleFilter--itemsList itemsList-custom-class">
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation is-active">
                    <input type="checkbox" id="filter_2000" checked="checked">
                    <label for="filter_2000">Year 2000 (30)</label>
                </a>           
            
      </div>
    </div>
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_1999" >
                    <label for="filter_1999">Year 1999 (12)</label>
                </a>           
            
      </div>
    </div>
  </div>
  <div
    class="asui-showMore showMoreContainer-custom-class"
    onClick={[Function anonymous]}
  >
    <div class="asui-showMore--more">
      <b>+ Show more</b>
    </div>
  </div>
</div>

`;

exports[`<MultipleFilterComponent /> should handleShowMore() and handleShowLess() 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-multipleFilter container-custom-class">
  <div class="asui-multipleFilter--top top-custom-class">Filter top header template:</div>
  <div class="asui-multipleFilter--itemsList itemsList-custom-class">
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_1999" >
                    <label for="filter_1999">Year 1999 (12)</label>
                </a>           
            
      </div>
    </div>
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_2000" >
                    <label for="filter_2000">Year 2000 (30)</label>
                </a>           
            
      </div>
    </div>
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_2001" >
                    <label for="filter_2001">Year 2001 (5)</label>
                </a>           
            
      </div>
    </div>
  </div>
  <div
    class="asui-showMore showMoreContainer-custom-class"
    onClick={[Function anonymous]}
  >
    <div class="asui-showMore--less">
      <b>- Show less</b>
    </div>
  </div>
</div>

`;

exports[`<MultipleFilterComponent /> should handleShowMore() and handleShowLess() 2`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-multipleFilter container-custom-class">
  <div class="asui-multipleFilter--top top-custom-class">Filter top header template:</div>
  <div class="asui-multipleFilter--itemsList itemsList-custom-class">
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_1999" >
                    <label for="filter_1999">Year 1999 (12)</label>
                </a>           
            
      </div>
    </div>
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_2000" >
                    <label for="filter_2000">Year 2000 (30)</label>
                </a>           
            
      </div>
    </div>
  </div>
  <div
    class="asui-showMore showMoreContainer-custom-class"
    onClick={[Function anonymous]}
  >
    <div class="asui-showMore--more">
      <b>+ Show more</b>
    </div>
  </div>
</div>

`;

exports[`<MultipleFilterComponent /> should render with custom props, and re-render with aggregations 1`] = `
Object {
  "aggregationField": "year_data",
  "applicationType": 4,
  "children": Array [],
  "classNames": Object {
    "container": "container-custom-class",
    "item": "item-custom-class",
    "itemsList": "itemsList-custom-class",
    "showMoreContainer": "showMoreContainer-custom-class",
    "top": "top-custom-class",
  },
  "dirty": true,
  "filterField": "year_id",
  "formatData": [Function],
  "limit": 2,
  "name": "year",
  "showMoreActive": true,
  "sortBy": Array [
    "_term",
    "desc",
  ],
  "target": "#some-target",
  "template": Object {
    "item": "
        <a class=\\"aggregation {{#isActive}}is-active{{/isActive}}\\">
            <input type=\\"checkbox\\" id=\\"filter_{{values.id}}\\" {{#isActive}}checked=\\"checked\\"{{/isActive}}>
            <label for=\\"filter_{{values.id}}\\">{{values.name}} ({{n}})</label>
        </a>           
    ",
    "showLess": "<b>- Show less</b>",
    "showMore": "<b>+ Show more</b>",
    "top": "Filter top header template:",
  },
}
`;

exports[`<MultipleFilterComponent /> should render with custom props, and re-render with aggregations 2`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-multipleFilter container-custom-class">
  <div class="asui-multipleFilter--top top-custom-class">Filter top header template:</div>
  <div class="asui-multipleFilter--itemsList itemsList-custom-class">
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_1999" >
                    <label for="filter_1999">Year 1999 (12)</label>
                </a>           
            
      </div>
    </div>
    <div
      class="asui-multipleFilter--item item-custom-class"
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_2000" >
                    <label for="filter_2000">Year 2000 (30)</label>
                </a>           
            
      </div>
    </div>
  </div>
  <div
    class="asui-showMore showMoreContainer-custom-class"
    onClick={[Function anonymous]}
  >
    <div class="asui-showMore--more">
      <b>+ Show more</b>
    </div>
  </div>
</div>

`;

exports[`<MultipleFilterComponent /> should render with default props 1`] = `
Object {
  "aggregationField": null,
  "applicationType": 8,
  "children": Array [],
  "classNames": Object {
    "container": "",
    "item": "",
    "itemsList": "",
    "showMoreContainer": "",
    "top": "",
  },
  "dirty": true,
  "filterField": "year_id",
  "formatData": [Function],
  "limit": 10,
  "name": "year",
  "showMoreActive": true,
  "sortBy": Array [
    "_term",
    "desc",
  ],
  "target": "#some-target",
  "template": Object {
    "item": "
        <a class=\\"aggregation {{#isActive}}is-active{{/isActive}}\\">
            <input type=\\"checkbox\\" id=\\"filter_{{values.id}}\\" {{#isActive}}checked=\\"checked\\"{{/isActive}}>
            <label for=\\"filter_{{values.id}}\\">{{values.name}} ({{n}})</label>
        </a>           
    ",
  },
}
`;

exports[`<MultipleFilterComponent /> should render with default props 2`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-multipleFilter ">
  <div class="asui-multipleFilter--itemsList ">
    <div
      class="asui-multipleFilter--item "
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_1999" >
                    <label for="filter_1999">Year 1999 (12)</label>
                </a>           
            
      </div>
    </div>
    <div
      class="asui-multipleFilter--item "
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_2000" >
                    <label for="filter_2000">Year 2000 (30)</label>
                </a>           
            
      </div>
    </div>
    <div
      class="asui-multipleFilter--item "
      onClick={[Function onClick]}
    >
      <div>
        
                <a class="aggregation ">
                    <input type="checkbox" id="filter_2001" >
                    <label for="filter_2001">Year 2001 (5)</label>
                </a>           
            
      </div>
    </div>
  </div>
</div>

`;
